profile_report.py

```diff
@@ -0,0 +1,31 @@
+from profiler import data
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("benchmark", type=str)
+parser.add_argument("path", type=str)
+
+args = parser.parse_args()
+
+# Processes the set of previously written logs
+
+threshold = 1.10  # Percent difference
+
+db = data.FileBasedProfileDB(args.path)
+actual_max_ts = 0
+dt = db.find(f"{args.benchmark}")
+c = 0
+
+if len(dt) >= 2:
+    exec_time = [0.0, 1.0]
+    last_two = dt[-2:]
+    exec_time[0] = last_two[0].user_time_sec
+    exec_time[1] = last_two[1].user_time_sec
+
+    if threshold * float(exec_time[1]) < float(exec_time[0]) or float(exec_time[1]) > threshold * float(exec_time[0]):
+        raise SystemExit(f"Potential performance degradation detected {exec_time[0]} vs {exec_time[1]}")
```

Comment:
If we allow for 10% degradation every time, we'll soon have unacceptable performance. Might we instead alert when a 10% degradation is detected compared to the _first_ benchmark?  If that makes sense, perhaps a larger degradation threshold is allowable.

Also, are the timings always for the same input data? If the data changes, we can't meaningfully compare the performance.

Reply:
I see your point about the 10% and also believe 10% compared to the first performance point makes more sense.

That's very true. I am not really sure how to handle the changes of data. I think that needs some brainstorming. One option is to consider new data as a new benchmark or reset the benchmark data when data changes significantly. However, none of these can be automated. What do you think?